### DISCUSSION TITLE
Finding documenation

### DISCUSSION THREAD

**Discussion by jeep** [gitkraken/vscode-gitlens#2306] [Feedback]
I'm looking for information about what alt-rightarrow does and I have no idea how to find out. The command is not well named as it is:
```
{
  "key": "alt+right",
  "command": "gitlens.key.alt+right",
  "when": "gitlens:key:alt+right"
}
```
That gives no indication of what the command does. I used google to try to find out and it revealed nothing. I can't find it in the documentation. How do I find out what command "gitlens.key.alt+right" does?

**Comment by eamodio** [comment]
That is a "internal" GitLens "command" and not intended to be used by end users. It's basically a "hack" to allow the VS Code keybinding system to allow GitLens to provide shortcut key based behaviors in different scenarios -- like in quick pick menus.

**Comment by jeep** [comment]
I don't understand... Alt-RightArrow is bound to this command. I hit Alt-RightArrow and nothing happens.... How do I find what is supposed to happen when I hit that key combination?

**Reply by eamodio** [reply]
Yeah, it's a dynamic binding -- it not to be used directly.

**Comment by d13** [comment]
> I hit Alt-RightArrow and nothing happens....

To be clear, the command doesn't do anything on it's own. Its used to capture keyboard events inside other features such as the quickpicks in this commit: https://github.com/gitkraken/vscode-gitlens/commit/7cdc9f7ee3587322d41180981e576b12555bd603

**Comment by jeep** [comment]
So two questions come up: 
1) First- my original question... where do I look for any documentation about keyboard shortcuts that might exist?
2) If I unbind alt-rightarrow (because I want to use that shortcut) will I break gitlens?

**Reply by eamodio** [reply]
Unfortunately there really isn't any documentation for the shortcut keys that GitLens adds other than what VS Code provides on the "Commands" section of the "Feature Contributions" tab on the extension page (which is definitely hard to use given the number of commands GitLens provides). And additionally unfortunate is that VS Code doesn't show any corresponding "when" condition, which are very important to know when a shortcut key will apply.

Another way to see the bound shortcuts with their when conditions is on the "Keyboard Shorcuts" page, `Open Keyboard Shorcuts` from the command palette. There if you type `gitlens.` you will see all the GitLens commands and their bound shortcuts.

In short, if you "unbind" `Alt+RightArrow` from GitLens, yes you will lose the ability to use `Alt+RightArrow` in GitLens' quick pick menus. But since those bindings only apply when certain conditions are met (e.g. you are focused in a GitLens quick pick), you can still bind `Alt+RightArrow` for other usages outside of that with any issue.

Hope that helps.

**Reply by jeep** [reply]
Thanks, this does answer the question. Though you pointed me to look at the Keyboard Shortcuts which is what spawned this question. ;)

It sounds like gitlens:key is supposed to be only in specific menus, which is good. I have no idea what that does or when I would hit it (I'm new to gitlens and there is apparently a lot more to it than what I've seen in the documentation.)

**Reply by eamodio** [reply]
As an example -- using the GitLens `Git Command Palette` (running `Git Command Palette` from the VS Code command palette), will start a "wizard-like" quick pick menu, where you can pick a git command and it will guide you through running it. You can use `Alt+LeftArrow` to go back from a later step to the previous step. And its in those quick pick menus, where those `gitlens.key.*` commands generally work. Although `gitlens.key.escape` is also used to cancel the File Annotations when they are active.
